youtube_dl/extractor/ard.py

```diff
@@ -395,73 +397,189 @@ class ARDBetaMediathekIE(ARDMediathekBaseIE):
+                    'alt_title':  alt_title
+                    }
+            )
+
+        elif episode_match:
+            episode_number = episode_match.group(1)
+            info.update(
+                {
+                    'season_number': int(0),
+                    'episode_number': int(episode_number),
+                    'alt_title':re.sub(r"\(\d+\)", "", title).replace("  ", "").strip()
+                }
+            )
+        else:
+            info.update(
+                {
+                    'alt_title': title
+                }
+            )
+
+        info.update(
+            {
+                'age_limit': age_limit,
+                'title': title,
+                'description': description,
+                'timestamp': unified_timestamp(player_page['widgets'][0].get('broadcastedOn')),
+                'series': try_get(player_page['widgets'][0], lambda x: x['show']['title']),
+                'channel': player_page['widgets'][0]['publicationService']['name'],
+                'channel_id': player_page['widgets'][0]['publicationService']['id'],
+                'channel_url': f"https://www.ardmediathek.de/{player_page['widgets'][0]['publicationService']['id']}",
+            }
+        )
         return info
+
+    def _real_extract_serie(self, video_id):
+        entries = []
+
+        page_number = 0
+        page_size = 100
+
+        while True:
+
+            widgets = self._download_json(
+                    f'https://api.ardmediathek.de/page-gateway/widgets/ard/asset/{video_id}',
+                    video_id,
+                    query={'pageSize':str(page_size), 'pageNumber':page_number}
+            )
+
+            for teaser in widgets['teasers']:
+                if 'EPISODE' == teaser['coreAssetType']:
+                    item = self._real_extract_video(teaser['id'])
+                    item['webpage_url'] = f"https://www.ardmediathek.de/video/{teaser['id']}"
+
+                    entries.append(item)
+
+            total = widgets['pagination']['totalElements']
+            if (page_number + 1) * page_size > total:
+                break
+            page_number += 1
+
+        return self.playlist_result(entries)
+
+    def _real_extract_named_serie(self, video_id, sender):
+
+        entries = []
+        page_size = 100
+
+        widgets = self._download_json(
+                f'https://api.ardmediathek.de/page-gateway/pages/{sender}/editorial/{video_id}',
+                video_id,
+                query={'pageSize': str(10), 'pageNumber': 0}
+        )['widgets']
+
+        for widget in widgets:
+            widget_id = widget['id']
+            page_number = 0
+
+            while True:
+                page_data = self._download_json(
+                        f'https://api.ardmediathek.de/page-gateway/widgets/{sender}/editorials/{widget_id}',
+                        video_id,
+                        query={'pageSize': page_size, 'pageNumber': page_number}
```

Comment:
Make loop condition explicit:
```suggestion
            total = 0
            while page_number * page_size <= total:
                page_data = self._download_json(
                    'https://api.ardmediathek.de/page-gateway/widgets/{0}/editorials/{1}'.format(sender, widget_id),
                    video_id, query={'pageSize': page_size, 'pageNumber': page_number}
```